Docker Desktop 4.7.0 release notes (#14505)
docker/docker.github.io: desktop/mac/release-notes/index.md

> chip](https://desktop.docker.com/mac/main/arm64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64){:
> .button .primary-btn }

### Security

> - Update Docker Engine to v20.10.14 to address [CVE-2022-24769](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-24769)

Reviewer: Should we add any of these CVEs to the Security announcements section in Docs - https://docs.docker.com/security/?

Author: Do we usually put Docker engine/upstream CVEs there @thaJeztah?

- Update containerd to v1.5.11 to address [CVE-2022-24769](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-24769)

### New

> - Add helper command and [instructions](https://docs.docker.com/desktop/mac/install/#install-from-the-command-line) for IT to install Docker Desktop remotely

Reviewer: Can we use relative links here?

- Add [Docker SBOM plugin v0.6.0](https://github.com/docker/sbom-cli-plugin/releases/tag/v0.6.0)

### Upgrades

> - [Docker Engine v20.10.13](https://docs.docker.com/engine/release-notes/#201013)

Reviewer 2: is it v20.10.14?

Author: yes!

- [Compose v2.4.1](https://github.com/docker/compose/releases/tag/v2.4.1)
- [Buildx 0.8.2](https://github.com/docker/buildx/releases/tag/v0.8.2)
- [containerd v1.5.11](https://github.com/containerd/containerd/releases/tag/v1.5.11)
- [Go 1.18](https://golang.org/doc/go1.18)

### Bug fixes and minor changes

>  - Deprecate macOS 10.14.

Reviewer: Yes. We've used a dedicated section to call out deprecated features - https://docs.docker.com/desktop/mac/release-notes/#deprecation

>  - Use `cri-dockerd` version 0.2.0 for new Kubernetes clusters.

Reviewer 3: We should say more about this, and in particular say that we are using it instead of dockershim, because lots of people will be worried (and/or spreading FUD) about the dockershim deprecation.

Author: @djs55: what do you suggest to put there to be a bit more precise?
(you did the original PR, so I suppose you know better than me why)

Reviewer 4: Perhaps
- Use [cri-dockerd](https://github.com/Mirantis/cri-dockerd) for Kubernetes clusters instead of "dockershim". Kubernetes containers run on the Docker engine as before. cri-dockerd allows Kubernetes to manage Docker containers via the standard [Container Runtime Interface](https://github.com/kubernetes/cri-api#readme), the same interface used to control other container runtimes. For more background, read ["The Future of Dockershim is cri-dockerd](https://www.mirantis.com/blog/the-future-of-dockershim-is-cri-dockerd/).

We could boost it to the "New" section at the top if we want to draw attention to it.

Author: good idea

>  - Fixed issue related to mysterious blank screen. Fixes [docker/for-mac#6134](https://github.com/docker/for-mac/issues/6134).

Reviewer 3: Delete "mysterious"

Reviewer: How about "Fixed an issue that sometimes caused Docker Desktop to display a blank white screen."

>  - Fixed a bug where the Registry Access Management policy would never be download if the first attempt to refresh fails.

Reviewer: Could you provide more info? What did the user try to refresh here?

Author: The automatic/background policy refresh was broken if it failed the first time, that's what I understand from the PR, right @djs55?

Reviewer 4: @ebriney that's correct. We download the policy
1. on login
2. periodically, to pick up changes made in Docker Hub

If a refresh failed, then we miscalculated the retry time and would never try again. So if the admin made a change to policy on Hub it would not propagate to the developer's machine automatically, it would only be downloaded if the developer logged out and then in again (or restarted the app).

Author: @usha-mandya perhaps something like 
```
 - Fixed a bug where the Registry Access Management policy was never refreshed after a failure.
```

>  - Fixed startup hang when osxfs is used for file sharing.

Reviewer: ```suggestion
 - Fixed an issue that caused Docker Desktop to become unresponsive during startup when osxfs is used for file sharing.
```

 - Improved login feedback.

> 

Author: Not sure what you're talking about!

> 

Reviewer 3: @djs55 knows.

> 

Reviewer 4: ```
- Fix a problem where gettimeofday() performance drops after waking from sleep when using hyperkit. Fixes [docker/for-mac#3455](https://github.com/docker/for-mac/issues/3455).
```

> chip](https://desktop.docker.com/mac/main/arm64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64){:
> .button .primary-btn }

### Security

> - Update Docker Engine to v20.10.14 to address [CVE-2022-24769](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-24769)

Reviewer: ```suggestion
- Update Docker Engine to v20.10.14 to address [CVE-2022-24769](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-24769){: target="_blank" rel="noopener" class="_"}
```

- Update containerd to v1.5.11 to address [CVE-2022-24769](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-24769)

### New

- IT Administrators can now install Docker Desktop remotely using the command line. For more information, see [Install Docker Desktop using the command line](../install.md#install-from-the-command-line).

> - Add [Docker SBOM plugin v0.6.0](https://github.com/docker/sbom-cli-plugin/releases/tag/v0.6.0)

Reviewer 3: Do we have any documentation about what this is or how people should use it?

Reviewer: We have a PR for SBOM updates - https://github.com/docker/docker.github.io/pull/14516

> - Add [Docker SBOM plugin v0.6.0](https://github.com/docker/sbom-cli-plugin/releases/tag/v0.6.0)

Reviewer: ```suggestion
- Add  the Docker Software Bill of Materials (SBOM) CLI plugin. The new CLI plugin enables users to generate SBOMs for Docker images. For more information, see [Docker SBOM](../../../engine/sbom/index.md).
```

> - Use [cri-dockerd](https://github.com/Mirantis/cri-dockerd) for new Kubernetes clusters instead of `dockershim`. Kubernetes containers run on the Docker engine as before. cri-dockerd allows Kubernetes to manage Docker containers via the standard [Container Runtime Interface](https://github.com/kubernetes/cri-api#readme), the same interface used to control other container runtimes. For more background, read [The Future of Dockershim is cri-dockerd](https://www.mirantis.com/blog/the-future-of-dockershim-is-cri-dockerd/).

Reviewer 3: I know this is already long, but I think it could benefit from another clause (and in the Windows section below):

... instead of `dockershim`. The change is transparent from the user's point of view, and Kubernetes ...

> - Use [cri-dockerd](https://github.com/Mirantis/cri-dockerd) for new Kubernetes clusters instead of `dockershim`. Kubernetes containers run on the Docker engine as before. cri-dockerd allows Kubernetes to manage Docker containers via the standard [Container Runtime Interface](https://github.com/kubernetes/cri-api#readme), the same interface used to control other container runtimes. For more background, read [The Future of Dockershim is cri-dockerd](https://www.mirantis.com/blog/the-future-of-dockershim-is-cri-dockerd/).

Reviewer: ```suggestion
- Use [cri-dockerd](https://github.com/Mirantis/cri-dockerd){: target="_blank" rel="noopener" class="_"}  for new Kubernetes clusters instead of `dockershim`. The change is transparent from the user's point of view and Kubernetes containers run on the Docker Engine as before. `cri-dockerd` allows Kubernetes to manage Docker containers using the standard [Container Runtime Interface](https://github.com/kubernetes/cri-api#readme){: target="_blank" rel="noopener" class="_"}, the same interface used to control other container runtimes. For more information, see [The Future of Dockershim is cri-dockerd](https://www.mirantis.com/blog/the-future-of-dockershim-is-cri-dockerd/){: target="_blank" rel="noopener" class="_"}.
```

### Upgrades

- [Docker Engine v20.10.14](https://docs.docker.com/engine/release-notes/#201014)
- [Compose v2.4.1](https://github.com/docker/compose/releases/tag/v2.4.1)
- [Buildx 0.8.2](https://github.com/docker/buildx/releases/tag/v0.8.2)
- [containerd v1.5.11](https://github.com/containerd/containerd/releases/tag/v1.5.11)
- [Go 1.18](https://golang.org/doc/go1.18)

### Bug fixes and minor changes
 - Fixed an issue that sometimes caused Docker Desktop to display a blank white screen. Fixes [docker/for-mac#6134](https://github.com/docker/for-mac/issues/6134).
 - Fixed a bug where the Registry Access Management policy would never be download if the first attempt to refresh fails.
 - Fixed an issue that caused Docker Desktop to become unresponsive during startup when osxfs is used for file sharing.
 - Fixed a problem where gettimeofday() performance drops after waking from sleep when using hyperkit. Fixes [docker/for-mac#3455](https://github.com/docker/for-mac/issues/3455).
 - Improved login feedback.

### Deprecation

>  - Deprecate macOS 10.14.

Reviewer: ```suggestion
 - Deprecate macOS 10.14. For macOS system requirements, see [Docker Desktop for Mac system requirements](../install.md#system-requirements).
```
